lib/dbservice_web/controllers/exam_controller.ex

```diff
@@ -0,0 +1,123 @@
+defmodule DbserviceWeb.ExamController do
+  use DbserviceWeb, :controller
+
+  import Ecto.Query
+  alias Dbservice.Repo
+  alias Dbservice.Exams
+  alias Dbservice.Exams.Exam
+
+  action_fallback(DbserviceWeb.FallbackController)
+
+  use PhoenixSwagger
+
+  alias DbserviceWeb.SwaggerSchema.Exam, as: SwaggerSchemaExam
+
+  def swagger_definitions do
+    Map.merge(
+      SwaggerSchemaExam.exams(),
+      SwaggerSchemaExam.exams()
+    )
+  end
+
+  swagger_path :index do
+    get("/api/exam")
+
+    parameters do
+      params(:query, :string, "The name of the exam", required: false, name: "name")
+    end
+
+    response(200, "OK", Schema.ref(:Exams))
+  end
+
+  def index(conn, params) do
+    query =
+      from(m in Exam,
+        order_by: [asc: m.id],
+        offset: ^params["offset"],
+        limit: ^params["limit"]
+      )
+
+    query =
+      Enum.reduce(params, query, fn {key, value}, acc ->
+        case String.to_existing_atom(key) do
+          :offset -> acc
+          :limit -> acc
+          atom -> from(u in acc, where: field(u, ^atom) == ^value)
+        end
+      end)
+
+    exam = Repo.all(query)
+    render(conn, "index.json", exam: exam)
+  end
+
+  swagger_path :create do
+    post("/api/exam")
+
+    parameters do
+      body(:body, Schema.ref(:Exam), "Exam to create", required: true)
+    end
+
+    response(201, "Created", Schema.ref(:Exam))
+  end
+
+  def create(conn, params) do
+    with {:ok, %Exam{} = exam} <- Exams.create_exam(params) do
+      conn
+      |> put_status(:created)
+      |> put_resp_header("location", Routes.exam_path(conn, :show, exam))
+      |> render("show.json", exam: exam)
+    end
+  end
+
+  swagger_path :show do
+    get("/api/exam/{examId}")
+
+    parameters do
+      examId(:path, :integer, "The id of the exam", required: true)
+    end
+
+    response(200, "OK", Schema.ref(:Exam))
+  end
+
+  def show(conn, %{"id" => id}) do
```

Comment:
there is inconsitency here.. you have mentioned "examId" above.. but using id here. this can lead to problems since show() will look for "id" instead of "examId"

Reply:
I don't think this makes a difference. Because in the swagger path, its just a representation. It will still look for the ID for the row.